packages/engine-cli/src/node-config-manager.ts

```diff
@@ -0,0 +1,140 @@
+import crypto from 'node:crypto';
+import esbuild from 'esbuild';
+import { runInContext, createContext } from 'node:vm';
+
+type BuildStats = {
+    error: Error | undefined;
+    currentValue: Promise<unknown[]> | unknown[];
+    dispose(): Promise<void>;
+    build?: ReturnType<typeof deferred<unknown[]>>;
+};
+
+export class NodeConfigManager {
+    constructor(private config: esbuild.BuildOptions) {}
+    runningBuilds = new Map<string, BuildStats>();
+    hashConfig(entryPoints: string[]) {
+        return hash(entryPoints.join(','));
```

Comment:
sort?

Reply:
nope. I don't think it's needed in this case. we pass a list of same config from the build so no variance.

Reply:
no sorting the order can will actually matter. but not for us.